src/port/portindex.tcl

```diff
@@ -321,12 +382,18 @@ try {
         close $oldfd
     }
     close $fd
+    if {$index_diff == 1} {
+        close $diff_fd
+    }
 }
 if {$exit_fail} {
     exit 1
 }
 
 file rename -force $tempportindex $outpath
+if {$index_diff == 1} {
+    file rename -force $tempchangedports $diffoutpath
```

Comment:
Currently what I could achieve is use the 'last modified' time of the previous portindex in the name of the file (telling us that this contains the changes since the last indexing). It is in UNIX time. Is it fine?

But, it would create multiple files with names `ChangedPorts-{timestamp}`. And since we are using timestamp of previous portindex in the name, I could not find a way to pick these files and clear them up (except for deleting all files that start with 'ChangedPorts')

Reply:
Not sure I understand. Why do we need to delete any file? We are certainly not reading any old diff file while creating the diff file. Even if a folder has multiple diff files, it's okay.